File: C/Algorithms/1902.c
char *largestOddNumber(char *num) {
  int set[10] = {0, 1, 0, 1, 0, 1, 0, 1, 0, 1}, len = strlen(num);
  for (int i = len - 1; i >= 0; i--) {
    if (set[num[i] - 48])
      break;
    num[i] = '\0';
  }
  return num;
}

File: Rust/Algorithms/1902.rs
impl Solution {
    pub fn largest_odd_number(mut num: String) -> String {
        let set = [0, 1, 0, 1, 0, 1, 0, 1, 0, 1];
        for c in num.clone().as_bytes().iter().rev() {
            if set[(c - 48) as usize] == 1 {
                break;
            }
            num.pop();
        }
        return num;
    }
}
